src/main/scala/exercise2/StackLike.scala

```diff
@@ -0,0 +1,52 @@
+package exercise2
+
+import scala.util.{Try}
+
+sealed trait StackLike[T] {
+  def push(elem: T): StackLike[T]
+
+  def pop(): Try[StackLike[T]] = this match {
+    case StackCons(_, tail) => Try(tail)
+    case StackEmpty() => Try(StackEmpty())
```

Comment:
`pop` on an empty stack should result in `Failure`, not `Success` with another empty stack.